src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,106 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+type Props = {
+  addMovies: (film: Movie) => void;
+};
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie: React.FC<Props> = ({ addMovies }) => {
+  const [movie, setMovie] = useState(null);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+
+  const loadMovie = () => {
+    getMovie(query)
+      .then(movieFromServer => {
+        if (movieFromServer.Error === 'Movie not found!') {
+          setError(true);
+          setMovie(null);
+        } else {
+          setMovie(movieFromServer);
+        }
+      });
+  };
+
+  const changeQuery = (event: { target: { value: string; }; }) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    setError(false);
+  };
+
+  const submitForm = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+    if (movie) {
+      addMovies(movie);
+      setQuery('');
+      setMovie(null);
+    }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+    setError(false);
+  };
+
+  return (
+    <>
+      <form
```

Comment:
would be great to move form in separate component

Reply:
> would be great to move form in separate component

Would you explain me why we should do this?  Form possesses all elements of FindeMovie.

Reply:
Because FindMovie has a lot of logic and it's a large component so it's better to split some login between components and make FindMovie more clear and readable